tests/unit/cli/test_cli_config_file.py

```diff
@@ -299,6 +303,63 @@ def test_save_command_line_args(self):
         self.assertNotIn("config_file", params.keys(), "Excluded member should not be saved to config")
         self.assertIn("some_param", params.keys(), "Param key should be saved to config file")
         self.assertIn("value", params.values(), "Param value should be saved to config file")
-        self.assertNotIn("some_other_param", params.keys(), "Param not passed in via COMMANDLINE should not be saved")
-        self.assertNotIn("none_param", params.keys(), "Param with None value should not be saved")
         mock_samconfig.flush.assert_called_once()  # everything should be flushed to config file
+
+    def test_dont_save_arguments_not_from_command_line(self):
+        mock_samconfig, mock_config_file = self._setup_mock_samconfig()
+        self.ctx = self._setup_context(
+            params={
+                "save_params": True,
+                "param_from_commandline": "value",
+                "param_not_from_commandline": "other_value",
+            },
+            parameter_source={
+                "save_params": ParameterSource.COMMANDLINE,
+                "param_from_commandline": ParameterSource.COMMANDLINE,
+                "param_not_from_commandline": ParameterSource.DEFAULT_MAP,
+            },
+        )
+
+        save_command_line_args_to_config(self.ctx, ["command"], "default", mock_samconfig)
+
+        self.assertIn(
+            "parameters",
+            mock_config_file.get("default", {}).get("command", {}).keys(),
+            "Parameters should be nested in config file",
+        )
+
+        params = mock_config_file["default"]["command"]["parameters"]
+
+        self.assertIn("param_from_commandline", params.keys(), "Param from COMMANDLINE should be saved to config file")
+        self.assertNotIn(
+            "param_not_from_commandline", params.keys(), "Param not passed in via COMMANDLINE should not be saved"
+        )
+
+    def test_dont_save_command_line_args_if_value_is_none(self):
+        mock_samconfig, mock_config_file = self._setup_mock_samconfig()
+        self.ctx = self._setup_context(
+            params={
+                "save_params": True,
+                "some_param": "value",
```

Comment:
nit: We could remove this since this case is already being tested and isn't checked here

Reply:
The case is there to make sure that the parameter itself actually gets written. If the only parameters being saved are `save_params` and `none_param` which don't get written to file, then the check for `default.command.parameters` fails, so the behaviour is indistinguishable from if the decorator just straight up doesn't run.